src/components/Menu/Menu.tsx

```diff
@@ -1,106 +1,90 @@
-  /** Menu position when open */
-  menuPosition?: MenuPositionAllowed;
-  /** The type of the button - defaults to "primary" */
-  buttonType?: ButtonTypes;
-  /** The name of the icon on the right area of menu button */
-  rightIconName?: AcceptedIconNames;
-  /** The name of the icon on the left area of menu button */
-  leftIconName?: AcceptedIconNames;
-  /** You can define the avatar properties here for src or letter if none then the user icon will be displayed if the object is not empty */
-  avatar?: {
-    src: string;
-    letter: string;
-    color?: AvatarColors;
-  };
-} & TestProps &
-  EventProps;
+  children?: React.ReactElement[] | React.ReactElement;
+  /** A callback that is being triggered when selection change */
+  onSelectionChange?: (e: Set<string>) => any;
+  /** The keys of the items that are disabled */
+  disabledKeys?: Set<string>;
+  /** The keys of the items that are selected */
+  selectedKeys?: Set<string>;
+  /** The size of the row */
+  rowSize?: ListRowSize;
+  /** The ref of the trigger element */
+  triggerRef: React.RefObject<any>;
+  /** Define if the menu is open or not */
+  isOpen?: boolean;
+  /** A callback that is being triggered when the menu is closed */
+  onClose?: (e: any) => any;
+  /** A callback that is being triggered when the menu is acting */
+  onAction?: (item: string) => any;
+  /** The selection mode of the menu */
+  selectionMode?: 'single' | 'multiple';
+} & TestProps;
 
-const Menu: React.FC<MenuProps> = (props) => {
-  const {
-    items,
-    onSelect,
-    buttonText = 'More',
-    menuPosition = 'left',
-    buttonType = 'primary',
-    rightIconName,
-    isDisabled = false,
-    leftIconName,
-    avatar,
-    dataTestId,
-  } = props;
-  const [isOpen, setIsOpen] = React.useState(false);
-  const theme = useTheme();
+const Menu: React.FC<MenuProps> = ({
+  selectionMode = 'single',
+  rowSize = 'normal',
+  children,
+  onSelectionChange = () => {},
+  disabledKeys = new Set(),
+  selectedKeys = new Set(),
+  triggerRef = null,
+  isOpen = false,
+  onClose = () => {},
+  onAction = () => {},
+  dataTestId,
+}) => {
+  const isCompact = rowSize === 'compact';
 
   return (
-    <ClickAwayListener onClick={() => setIsOpen(false)}>
-      <div data-testid={dataTestId}>
-        <PositionInScreen
-          isVisible={isOpen}
-          offsetY={8}
-          parent={
-            <Button
-              type={buttonType}
-              onClick={() => setIsOpen(!isOpen)}
-              isDisabled={isDisabled}
-              iconRightName={rightIconName}
-              iconLeftName={leftIconName}
-              avatar={!isEmpty(avatar) ? { src: avatar?.src, label: avatar?.letter } : undefined}
-            >
-              <span>{buttonText}</span>
-            </Button>
-          }
+    <MenuTrigger isOpen={isOpen} onOpenChange={onClose}>
+      <Popover
+        triggerRef={triggerRef}
+        css={css`
+          min-width: 150px;
+          background: #ffffff;
```

Comment:
maybe move this to Menu.style.ts and use rem for min-width and global colors (or ideally menu tokens) for the background

Reply:
fixed https://github.com/Orfium/orfium-ictinus/pull/747/commits/dd9af368f8341e0533aca239112f1d4e39a07471